dask_expr/_rolling.py

```diff
@@ -49,7 +74,19 @@ def kwargs(self):
 
     def _simplify_up(self, parent):
         if isinstance(parent, Projection):
-            return type(self)(self.frame[parent.operand("columns")], *self.operands[1:])
+            by = self.groupby_kwargs.get("by", []) if self.groupby_kwargs else []
+            by_columns = by if not isinstance(by, Expr) else []
+            columns = sorted(set(parent.columns + by_columns))
+            if columns == self.frame.columns:
+                return
+            if self.groupby_kwargs is not None:
+                return type(parent)(
+                    type(self)(self.frame[columns], *self.operands[1:]),
+                    *parent.operands[1:],
+                )
```

Comment:
I'm honestly not even sure what this is doing and it does feel like we're duplicating work but this is what I found for the groupby implementations and this seems to work.

Reply:
This drops all columns except the parent columns and the by columns from the df before the groupby and then drops the by column after the groupby if not in parent.columns

Yes this is needed for groupby to avoid dropping too many columns